src/main/java/duke/exceptions/DukeException.java

```diff
@@ -0,0 +1,70 @@
+package duke.exceptions;
+
+import duke.enums.ErrorCodes;
+
+public class DukeException extends Exception {
+    private ErrorCodes errorCode;
+    private String errorInfo;
+
+    /**
+     * DukeException constructor.
+     *
+     * @param errorCode An ErrorCode enum representing the error type.
+     */
+    public DukeException(ErrorCodes errorCode) {
+        super();
+        this.errorCode = errorCode;
+    }
+
+    /**
+     * DukeException constructor with additional error information.
+     *
+     * @param errorCode An ErrorCode enum representing the error type.
+     * @param errorInfo A string containing any additional information.
+     */
+    public DukeException(ErrorCodes errorCode, String errorInfo) {
+        super();
+        this.errorCode = errorCode;
+        this.errorInfo = errorInfo;
+    }
+
+    @Override
+    public String getMessage() {
+        String message = "";
+        switch (errorCode) {
+        case UNKNOWN_COMMAND:
```

Comment:
I like the way you organised your exceptions command messages. Keep up good work!